src/Contracts/Models/CartDeduction.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tipoff\Checkout\Contracts\Models;
+
+use Brick\Money\Money;
+
+interface CartDeduction
+{
+    public static function findDeductionByCode(string $code): ?CartDeduction;
+
+    public static function calculateCartDeduction(CartInterface $cart): Money;
+
+    public static function markCartDeductionsAsUsed(CartInterface $cart): void;
+
+    public function applyToCart(CartInterface $cart);
+
+    /**
+     * @param CartInterface $cart
+     * @return array|string[]
+     */
+    public function getCodesForCart(CartInterface $cart): array;
+}
```

Comment:
Defines a common concept of "CartDeduction" that both Discount and Voucher will implement.  This allows cart to handle these (and any future coded deduction) in the same way.